XenAdmin/TabPages/AlertSummaryPage.cs

```diff
@@ -585,24 +584,27 @@ private void tsmiDismissSelected_Click(object sender, EventArgs e)
         private void AlertsCollectionChanged(object sender, CollectionChangeEventArgs e)
         {
             Program.AssertOnEventThread();
-            if (e.Element == null)
-            {
-                // We take the null element to mean there has been a batch remove
-                Rebuild();
-                return;
-            }
-            Alert a = e.Element as Alert;
+
             switch (e.Action)
             {
                 case CollectionChangeAction.Add:
                     Rebuild(); // rebuild entire alert list to ensure filtering and sorting
                     break;
                 case CollectionChangeAction.Remove:
-                    RemoveAlertRow(a);
+
+                    var a = e.Element as Alert;
+                    if (a != null)
+                        RemoveAlertRow(a);
+                    else
+                    {
+                        var range = e.Element as List<Alert>;
+                        if (range != null)
+                            Rebuild();

```

Comment:
I wonder if it's worth checking if there was just one element in the list, and in this case call RemoveAlertRow?